ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/MainActivity.kt

```diff
@@ -73,44 +76,77 @@ class MainActivity : ComponentActivity() {
 @Composable
 fun WearApp(watchRepository: WatchRepository) {
     WearAppTheme {
-        // The state for the ScalingLazyColumn in other screen is hoisted to this level, so
-        // the Scaffold can properly place the position indicator (also known as the
-        // scroll indicator). We use it for various other things (like hiding the time when
-        // the user is scrolling).
-        //
-        // This current solution works for one scrolling type Composable, that is, we only
-        // have one Composable capable of scrolling.
+        // The state for any scrollable content in the composable screens is hoisted to this level,
+        // so the Scaffold can properly place the position indicator (also known as the
+        // scroll indicator).
+        // In our case, the watch list screen uses a ScalingLazyColumn and the watch details
+        // screen uses a Column with scrolling enabled.
+        // We also use the states for various other things (like hiding the time when the user is
+        // scrolling).
+        // ----------
+        // State for the ScalingLazyColumn (watch list screen)
         val scalingLazyListState: ScalingLazyListState = rememberScalingLazyListState()
+        // State for the Column's vertical scroll modifier (watch detail screen).
+        val watchDetailScrollState: ScrollState = rememberScrollState()
```

Comment:
One interesting effect of this setup: Because this `ScrollState` is hoisted outside of the navigation, this state will be preserved even when the user leaves the detail to go back. That causes the following:

- User navigates to the list
- User navigates to detail A
- User scrolls to the bottom of detail A
- User navigates back
- User navigates to detail B
- User is now already scrolled on detail B with detail A's scroll state.

Avoiding that is a bit tricky, and in the general case with nested navigation (say, if you could navigate from a detail to another detail view) since we can't just have a nice, fixed set of hoisted states.

Reply:
Yeah, not ideal... I guess I could reset the scroll state value to 0 for each detail screen launch or do some sort of logic where if the detail screen isn't equal to the previous detail screen that was used it resets.

Reply:
Yeah, either of those should work, although it's unfortunate it doesn't solve the general case... It's unfortunate that `NavController.backQueue` is hidden.

Reply:
I reworked it to use a remembered map and that works pretty well. PTAL.